packages/react-sdk/docusaurus/docs/React/06-ui-cookbook/06-lobby-preview.mdx

```diff
@@ -3,5 +3,248 @@ title: Lobby Preview
+
+**Device selector example**
+
+The selectors can be thought of as dropdowns in our example.
+
+```tsx
+import {
+  useAudioInputDevices,
+  useAudioOutputDevices,
+  useMediaDevices,
+  useVideoDevices,
+} from '@stream-io/video-react-sdk';
+
+type DeviceSelectorProps = {
+  devices: MediaDeviceInfo[];
+  deviceKind: MediaDeviceKind;
+  selectedDeviceId?: string;
+};
+
+export const DeviceSelector = ({ devices, deviceKind, selectedDeviceId }: DeviceSelectorProps) => {
+  // highlight-next-line
+  const { switchDevice } = useMediaDevices();
+
+  return (
+    <div className="selector">
+      {devices.map((device) => {
+        return (
+          <div
+            className={`option${selectedDeviceId === device.deviceId ? 'option--selected' : ''}`}
+            key={device.deviceId}
+            // highlight-next-line
+            onClick={ () => switchDevice(deviceKind, device.deviceId) }
+          >
+            {device.label}
+          </div>
+        )
+      })}
+    </div>
+  );
+}
+
+export const AudioInputDeviceSelector = () => {
+  const { selectedAudioInputDeviceId } = useMediaDevices();
+  // maintains an up-to-date array of MediaDeviceInfo objects of kind "audioinput"
+  const audioInputDevices = useAudioInputDevices();
+
+  return (
+    <DeviceSelector
+      devices={ audioInputDevices }
+      deviceKind="audioinput"
+      selectedDeviceId={ selectedAudioInputDeviceId }
+    />
+  );
+}
+
+export const AudioOutputDeviceSelector = () => {
+  const {
+    isAudioOutputChangeSupported,
+    selectedAudioOutputDeviceId,
+  } = useMediaDevices();
+  // maintains an up-to-date array of MediaDeviceInfo objects of kind "audiooutput"
+  const audioOutputDevices = useAudioOutputDevices();
+
+  if (!isAudioOutputChangeSupported) return null;
+
+  return (
+    <DeviceSelector
+      devices={ audioOutputDevices }
+      deviceKind="audiooutput"
+      selectedDeviceId={ selectedAudioOutputDeviceId }
+    />
+  );
+}
+
+export const VideoInputDeviceSelector = () => {
+  const { selectedVideoDeviceId } = useMediaDevices();
+  // maintains an up-to-date array of MediaDeviceInfo objects of kind "videoinput"
+  const videoDevices = useVideoDevices();
+
+  return (
+    <DeviceSelector
+      devices={ videoDevices }
+      deviceKind="videoinput"
+      selectedDeviceId={ selectedVideoDeviceId }
+    />
+  );
+}
+```
+
+### Participants in a call
+We can retrieve the list of members, that already joined the call (participants), by inspecting the call metadata object (`callMetadata.session.participants`). The object is provided and maintained up-to-date by [`useCallMetadata`](../call-engine/hooks-and-contexts#usecallmetadata) hook.
+
+```tsx
+import { Avatar, useCallMetadata } from '@stream-io/video-react-sdk';
+
+export const ParticipantsPreview = () => {
+  const callMetadata = useCallMetadata();
+
+  if (
+    !(
+      callMetadata?.session?.participants &&
+      callMetadata?.session?.participants.length
+    )
+  )
+    return null;
```

Comment:
I'm considering whether we should introduce a `useCallCurrentSession()` hook... I think accessing the call's current session might be a common use case. What do you think?

Reply:
If session is always updated with metadata, I would probably not introduce a new metadata hook in order to avoid having too many hooks.

Reply:
At least for now